robolectric/src/test/java/org/robolectric/shadows/ShadowNotificationManagerTest.java

```diff
@@ -93,4 +97,32 @@ public void testCancelAll() throws Exception {
     assertNull(shadowOf(notificationManager).getNotification(null, 1));
     assertNull(shadowOf(notificationManager).getNotification(null, 31));
   }
+
+  @Test
+  public void testGetActiveNotifications() throws Exception {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+      // This API was added in M.
+      return;
+    }
+    notificationManager.notify(1, notification1);
+    notificationManager.notify(31, notification2);
+
+    StatusBarNotification[] statusBarNotifications =
+        shadowOf(notificationManager).getActiveNotifications();
+    assertEquals(2, statusBarNotifications.length);
+    boolean hasNotification1 = false;
+    boolean hasNotification2 = false;
+    for (StatusBarNotification notification : statusBarNotifications) {
```

Comment:
Maybe switch the for loop out of the test and have a method getNotificationWithId(int id) that returns the notification or null if not present. I think the test reads better without loops in.